### PR TITLE
Updated statistical analysis parameters in API methods

### PR DESCRIPTION
**Pull request by cadeyansys**
As part of the statistical analysis input parameter changes (now adding a user interface), updated the statistical analysis API methods to the new stats_params structure rather than a part of properties.

### CODE REVIEW

File: src/ansys/dynamicreporting/core/utils/report_objects.py [hunk 1]
```diff
@@ -3473,49 +3473,129 @@ class statisticalREST(GeneratorREST):
...
 
     def __init__(self):
         super().__init__()
 
-    @property
-    def analysis_type(self) -> str:
-        return self.get_property().get("analysis_type")
+    def get_analysis_type(self):
+        if "stats_params" in json.loads(self.params):
+            if "analysis_type" in json.loads(self.params)["stats_params"]:
```
Comment (viseshrp) [inline src/ansys/dynamicreporting/core/utils/report_objects.py:3486]: Little improvement: remove the nesting and join the conditions together with an `and`

File: src/ansys/dynamicreporting/core/utils/report_objects.py [hunk 2]
```diff
@@ -3473,49 +3473,129 @@ class statisticalREST(GeneratorREST):
...
+                return json.loads(self.params)["stats_params"]["analysis_type"]
+        return ""
 
-    @analysis_type.setter
-    def analysis_type(self, value: str) -> None:
-        props = self.get_property()
-        props["analysis_type"] = value
-        self.set_property(props)
+    def set_analysis_type(self, value=""):
+        if type(value) is not str:
```
Comment (viseshrp) [inline src/ansys/dynamicreporting/core/utils/report_objects.py:3491]: Check for instances of str instead: `if not isinstance(value, str)`. More pythonic.

File: src/ansys/dynamicreporting/core/utils/report_objects.py [hunk 3]
```diff
@@ -3473,49 +3473,129 @@ class statisticalREST(GeneratorREST):
...
+                "Error: input does not belong to the set of supported analysis types. Must be one of [{a_types}].".format(
+                    a_types=", ".join(analysis_types)
+                )
+            )
+        d = json.loads(self.params)
+        if "stats_params" not in d:
+            d["stats_params"] = {}
+        d["stats_params"]["analysis_type"] = value
+        self.params = json.dumps(d)
+        return
```
Comment (viseshrp) [inline src/ansys/dynamicreporting/core/utils/report_objects.py:3506]: The `return` is implied and can be removed.

File: src/ansys/dynamicreporting/core/utils/report_objects.py [hunk 4]
```diff
@@ -3473,49 +3473,129 @@ class statisticalREST(GeneratorREST):
...
 
-    @property
-    def predictor_variables(self):
-        return self.get_property().get("predictor_variables")
+    def get_table_name(self):
+        if "stats_params" in json.loads(self.params):
```
Comment (viseshrp) [inline src/ansys/dynamicreporting/core/utils/report_objects.py:3509]: Same as above.

File: src/ansys/dynamicreporting/core/utils/report_objects.py [hunk 5]
```diff
@@ -3473,49 +3473,129 @@ class statisticalREST(GeneratorREST):
...
+                return json.loads(self.params)["stats_params"]["table_name"]
+        return ""
 
-    @predictor_variables.setter
-    def predictor_variables(self, value):
-        props = self.get_property()
-        props["predictor_variables"] = value
-        self.set_property(props)
+    def set_table_name(self, value=""):
+        if type(value) is not str:
```
Comment (viseshrp) [inline src/ansys/dynamicreporting/core/utils/report_objects.py:3515]: Same as above

File: src/ansys/dynamicreporting/core/utils/report_objects.py [hunk 6]
```diff
@@ -3473,49 +3473,129 @@ class statisticalREST(GeneratorREST):
...
+        return
+
+    def get_response_variables(self):
+        if "stats_params" in json.loads(self.params):
+            if "response_variables" in json.loads(self.params)["stats_params"]:
+                return json.loads(self.params)["stats_params"]["response_variables"]
+        return ""
+
+    def set_response_variables(self, value=""):
+        if type(value) is not list:
```
Comment (viseshrp) [inline src/ansys/dynamicreporting/core/utils/report_objects.py:3568]: `if not isinstance(value, list)`

**Review by viseshrp — APPROVED** [review]
Mostly fine, just some merge code improvements.

File: src/ansys/dynamicreporting/core/utils/report_objects.py (hunk 6 above)
Comment (cadeyansys) [reply src/ansys/dynamicreporting/core/utils/report_objects.py:3568]: Hello Visesh, thank you so much your suggestion, will update this part

File: src/ansys/dynamicreporting/core/utils/report_objects.py (hunk 3 above)
Comment (cadeyansys) [reply src/ansys/dynamicreporting/core/utils/report_objects.py:3506]: Gotcha, thank you Visesh, didn't consider how this is implied before

File: src/ansys/dynamicreporting/core/utils/report_objects.py (hunk 1 above)
Comment (cadeyansys) [reply src/ansys/dynamicreporting/core/utils/report_objects.py]: Thank you so much for the suggestions Visesh 😃 Updated code with the suggested improvements!

**Pull request merged**